### ISSUE TITLE
Unable to install ms-python.python, ms-toolsai.jupyter not compatible with current VS Code version, 1.58.0

### ISSUE THREAD

**Issue by scaeuus** [microsoft/vscode#133043]
Issue Type: <b>Bug</b>

Steps to Reproduce:

1. Just after a clean install, launched VS Code 
2. Searched for ms-python.python, clicked 'Install'
3. Returns 'Unable to install ms-toolsai.jupyter extension because it is not compatible with the current version of VS Code (version 1.58.0).' 

VS Code version: Code - OSS 1.58.0 (2d23c42a936db1c7b3b06f918cde29561cc47cd6, 2021-07-12T23:51:36.303Z)
OS version: Linux x64 5.13.13-1-MANJARO
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4710MQ CPU @ 2.50GHz (8 x 900)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 0, 0|
|Memory (System)|15.56GB (9.34GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|x11|
</details>Extensions: none

**Comment by iohzrd** [comment]
I'm experiencing the same thing, very similar environment... Manjaro...

**Comment by sandy081** [comment]
> Code - OSS 1.58.0 

I do not think we support extension installation on OSS build. If you got this installer through another source, please file an issue against that.

**Comment by bneijt** [comment]
@sandy081 Extension installation is supported in the OSS build, I've been using that since I started using code.

Could you please check if you are dropping support for the OSS build officially or whether this is just a hunch/rumor?

**Comment by scaeuus** [comment]
@sandy081 I've got the installer straight from OSS Build (or tried to, since I couldn't end installation because of the bug I reported). Installing extensions from Marketplace it's been always available. (Reproduced both in Manjaro Gnome and Manjaro KDE)
**Fresh install, this time on Manjaro KDE**
![image](https://user-images.githubusercontent.com/57377730/133658429-0840a51f-487b-4879-97c6-6b27871edddd.png)

**Taking a look at Marketplace**
![image](https://user-images.githubusercontent.com/57377730/133658675-cf022076-990b-4f07-857c-00e52745c238.png)

**Going for installation**
![image](https://user-images.githubusercontent.com/57377730/133658841-bf9ae8c1-516a-471b-9c12-9829c7544f51.png)

**Not allowed because...**
![image](https://user-images.githubusercontent.com/57377730/133658993-06e44ba1-c088-4f94-92d7-52a002069d22.png)

**Comment by scaeuus** [comment]
> @sandy081 Extension installation is supported in the OSS build, I've been using that since I started using code.
> 
> Could you please check if you are dropping support for the OSS build officially or whether this is just a hunch/rumor?

@iohzrd, @bneijt have you considered trying out a different version? https://wiki.archlinux.org/title/Visual_Studio_Code 
I've just managed to install VS Code and its Python(Pylance) extension.

![image](https://user-images.githubusercontent.com/57377730/133711629-50051779-4347-45ba-b99d-35f698cf4594.png)

**Comment by sandy081** [comment]
https://wiki.archlinux.org/title/Visual_Studio_Code says following 

> Visual Studio Code is a cross-platform proprietary text editor (licensed under an End-User License Agreement) developed by Microsoft, built on top of a free and open-source project named Codium (licensed under the MIT License),

It is built on top of Codium which is different. So please file issue against Codium project.

**Comment by bneijt** [comment]
> https://wiki.archlinux.org/title/Visual_Studio_Code says following
> 
> > Visual Studio Code is a cross-platform proprietary text editor (licensed under an End-User License Agreement) developed by Microsoft, built on top of a free and open-source project named Codium (licensed under the MIT License),
> 
> It is built on top of Codium which is different. So please file issue against Codium project.

Oh, I guess the wiki is wrong there. Codium is a community rebrand/rebuild of vscode which is an open source version from Microsoft hosted here in this repo. (Think Iceweasel vs Firefox). Then there is also the proprietary binary Visual Studio Code, which is based on this open source version but closed by a Microsoft license. You can only get the binary there, source is protected.

The only issue here is that you need at least version 1.60 to allow for the marketplace plugin to install. This is not a Codium issue as the above mentioned the user is not running Codium but the built from source vscode from this repository.

To summarize:
- No need to post bug anywhere else
- You need to upgrade to at least 1.60 to be able to use ms-python plugin
- The whole closed/open/combined licensing forest is confusing for a lot of people ;)

**Comment by scaeuus** [comment]
@sandy081 and @bneijt thanks for clarifying this. 
>     * The whole closed/open/combined licensing forest is confusing for a lot of people ;)
You're absolutly right @bneijt . Sorry for my confusion.

**Comment by scaeuus** [comment]
> @sandy081 Extension installation is supported in the OSS build, I've been using that since I started using code.
> 
> Could you please check if you are dropping support for the OSS build officially or whether this is just a hunch/rumor?

@iohzrd, @bneijt have you considered trying out a different version? https://wiki.archlinux.org/title/Visual_Studio_Code 
I've just managed to install VS Code and its Python(Pylance) extension.

![image](https://user-images.githubusercontent.com/57377730/133711629-50051779-4347-45ba-b99d-35f698cf4594.png)

**Comment by tamuratak** [comment]
The Code OSS build is flagged as out-of-date. It seems not to be maintained anymore. See https://archlinux.org/packages/community/x86_64/code/

**Comment by scaeuus** [comment]
You're right.

   - Source Files
   <https://github.com/archlinux/svntogit-community/tree/packages/code/trunk>
   / View Changes
   <https://github.com/archlinux/svntogit-community/commits/packages/code/trunk>
   - Bug Reports <https://bugs.archlinux.org/?project=5&string=code> / Add
   New Bug
   <https://bugs.archlinux.org/newtask?project=5&product_category=33&item_summary=%5Bcode%5D+PLEASE+ENTER+SUMMARY>
   - Search Wiki
   <https://wiki.archlinux.org/title/Special:Search?search=code> / Manual
   Pages <https://man.archlinux.org/listing/code>
   - Security Issues <https://security.archlinux.org/package/code>
      - *Flagged out-of-date on 2021-07-13*
   - Download From Mirror
   <https://archlinux.org/packages/community/x86_64/code/download>
